File: GoF/comportamentais/template_method/python/pdf_dataminer.py
from dataminer import DataMiner


class PdfDataMiner(DataMiner):
    def open_file(self, path):
        print("Arquivo PDF aberto")
        return "Arquivo PDF"

    def extract_data(self, file):
        return "Dados do arquivo PDF"

    def parse_data(self, raw_data):
        return raw_data + " | Dados PDF tratados"

    def close_file(self, file):
        print("Arquivo PDF fechado")

File: GoF/comportamentais/template_method/java/PdfDataMiner.java
public class PdfDataMiner extends DataMiner{
    @Override
    public String openFile(String path){
        System.out.println("Arquivo PDF aberto");
        return "Arquivo PDF";
    }

    @Override
    public String extractData(String file){
        return "Dados do arquivo PDF";
    }

    @Override
    public String parseData(String rawData){
        return rawData + " | Dados PDF tratados";
    }

    @Override
    public void closeFile(String file){
        System.out.println("Arquivo PDF fechado");
    }

}